src/modules/i18n/containers/__tests__/I18nProvider.spec.js

```diff
@@ -149,32 +153,21 @@ describe.skip('I18nProvider', () => {
       const originalGetSelectedFonts = I18nProvider.getSelectedFonts
       // $FlowFixMe
       I18nProvider.getSelectedFonts = jest.fn(I18nProvider.getSelectedFonts)
-      i18n.changeLanguage = jest.fn()
+      i18n.changeLanguage = jest.fn(i18next.changeLanguage)
 
-      component.instance().setLanguage(expectedLanguage)
+      await component.instance().setLanguage(expectedLanguage)
 
       // $FlowFixMe
       expect(document.documentElement.lang).toEqual(expectedLanguage)
       expect(i18n.changeLanguage).toHaveBeenCalledWith(expectedLanguage)
       expect(I18nProvider.getSelectedFonts).toHaveBeenCalledWith(expectedLanguage)
       expect(component.state()).toEqual({
         language: expectedLanguage,
+        i18nLoaded: true,
         fonts: { lateef: true, openSans: true, raleway: true }
       })
       // $FlowFixMe
       I18nProvider.getSelectedFonts = originalGetSelectedFonts
     })
   })
-
-  it('should add direction style depending on language', () => {
```

Comment:
The test below runs this test succesfully and shows perfectly why setUiDirection is quite difficult to test. Maybe we should consider moving setUiDirection and setLanguage from componentDidUpdate to ComponentWillUpdate, but I am sure there was some intention behind componentDidUpdate.
```javascript
it('should add direction style depending on language', async () => {
  const mockSetUiDirection = jest.fn(setUIDirection)
  const component = mount(<I18nProvider language='en' setUiDirection={mockSetUiDirection}>
    <div />
  </I18nProvider>)
  expect(component.find('div').at(0).prop('style').direction).toEqual('ltr')
  component.setProps({ language: 'ar' })
  component.update()
  await new Promise(resolve => setTimeout(resolve, 200))
  component.update()
  expect(mockSetUiDirection).toHaveBeenCalledWith('rtl')
  expect(component.find('div').at(0).prop('style').direction).toEqual('rtl')
})
```

Reply:
And without the `setTimeout` it does not work?

Reply:
componentWillUpdate is deprecated unfortunately: https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate

Reply:
It would also be possible to use `wait-for-expect` (https://github.com/TheBrainFamily/wait-for-expect) to omit using a error prone timeout.

Reply:
> componentWillUpdate is deprecated unfortunately: https://reactjs.org/docs/react-component.html#unsafe_componentwillupdate

did not know that. Is there another way to avoid calling update twice here? 

> It would also be possible to use `wait-for-expect` (https://github.com/TheBrainFamily/wait-for-expect) to omit using a error prone timeout.

I will try that.

Reply:
In fact I'm not sure why we need `update` anyway... according to the enzyme docs `setProps` should already force a re-render: https://enzymejs.github.io/enzyme/docs/api/ShallowWrapper/setProps.html
The second `update` should also be not needed AFAIK since in theory it should automatically rerender after `setLanguage`/`setUiDirection` is called :thinking:

Reply:
Somehow the issue with the timimg was related to changeLanguage in itself. The state update did not trigger the ui-direction to change. I am actually not quite sure what the issue was, probably something with the I18NextProvider to be unmounted or sth, but I am not sure. However it works now, as I moved the code from then(..) into the callback of changeLanguage.

Reply:
The other update is still needed though?

Reply:
nope.

Reply:
But it's still there :thinking:

Reply:
wait. what exactly do you mean? I thought you meant the `component.update()` in line 183. And there you do need to call update once otherwise `componentDidUpdate` will not be triggered

Reply:
Yes, but you wrote `nope`, its not needed anymore :grin:

Reply:
Ah. I thought you ment the second update in the code snippet above. To clarify: We do need to call update at least once.